src/rules.jl

```diff
@@ -100,15 +100,11 @@ end
 # trinary #
 #---------#
 
-#=
-
 @define_diffrule Base.muladd(x, y, z) = :($y), :($x), :(one($z))
 @define_diffrule Base.fma(x, y, z)    = :($y), :($x), :(one($z))
 
 @define_diffrule Base.ifelse(p, x, y) = false, :($p), :(!$p)
```

Comment:
Ideally, the first argument should be marked as non-differentiable I think. But AFAICT this is not supported by DiffRules (or rather handled in a crude way by returning `NaN` for both non-differentiable arguments and non-implemented derivatives)?

Reply:
Yes I think I wasn't sure a year ago what this package did, but I see some other NaN. Which as you say seems a reasonable way to encode non-implemented / non-diff. 

```suggestion
@define_diffrule Base.ifelse(p, x, y) = NaN, :($p), :(!$p)
```